components/dashboard/src/settings/ProfileInformation.tsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { User } from "@gitpod/gitpod-protocol";
+import { hoursBefore, isDateSmaller } from "@gitpod/gitpod-protocol/lib/util/timeutil";
+import React, { useContext, useState } from "react";
+import CodeText from "../components/CodeText";
+import Modal from "../components/Modal";
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
+
+export namespace ProfileState {
+    export interface ProfileState {
+        name: string;
+        email: string;
+        company?: string;
+        avatarURL?: string;
+    }
+
+    export function getProfileState(user: User): ProfileState {
+        return {
+            name: User.getName(user!) || "",
+            email: User.getPrimaryEmail(user!) || "",
+            company: user?.additionalData?.companyName,
+            avatarURL: user?.avatarUrl,
+        };
+    }
+
+    export function setProfileState(user: User, profileState: ProfileState): User {
+        user.fullName = profileState.name;
+        user.avatarUrl = profileState.avatarURL;
+
+        if (!user.additionalData) {
+            user.additionalData = {};
+        }
+        user.additionalData.emailAddress = profileState.email;
+        user.additionalData.companyName = profileState.company;
+        user.additionalData.lastUpdatedDetailsNudge = new Date().toISOString();
+
+        return user;
+    }
+
+    export function hasChanges(before: ProfileState, after: ProfileState) {
+        return (
+            before.name !== after.name ||
+            before.email !== after.email ||
+            before.company !== after.company ||
+            before.avatarURL !== after.avatarURL
+        );
+    }
+
+    function shouldNudgeForUpdate(user: User): boolean {
+        if (!user.additionalData) {
+            // never updated profile information and account is older tha 24 hours (i.e. ask on second day).
+            return !isDateSmaller(hoursBefore(new Date().toISOString(), 24), user.creationDate);
+        }
+        // if the profile wasn't updated for 12 months ask again.
+        return !(
+            !!user.additionalData.lastUpdatedDetailsNudge &&
+            isDateSmaller(hoursBefore(new Date().toISOString(), 24 * 365), user.additionalData.lastUpdatedDetailsNudge)
+        );
+    }
+
+    /**
+     * @param state
+     * @returns error message or empty string when valid
+     */
+    export function validate(state: ProfileState): string {
+        if (state.name.trim() === "") {
+            return "Name must not be empty.";
+        }
+        if (state.email.trim() === "") {
+            return "Email must not be empty.";
+        }
+        if (!/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(state.email.trim())) {
+            return "Please enter a valid email.";
```

Comment:
**issue(non-blocking)**: This does not safely filter emails with `+` like `user+suffix@company.com`.